### ISSUE TITLE
TSQL exec call replaces = with IS when parameter is set to null

### ISSUE THREAD

**Issue by fdw** [sqlfluff/sqlfluff#2295]
The formatting for stored procedures produces invalid SQL if one parameter is set to `null`.

```
exec function
  @param1 = 'blah',
  @param2 = null;
```

## Expected Behaviour
Ideally, the code would be formatted (#1861). Otherwise, it should be left as-is.

## Observed Behaviour
`@param2`'s `=` is replaced by `IS`, which is not valid TSQL anymore:
```
exec function
@param = 'blah';
@param IS null;
```

## Steps to Reproduce
see above

## Dialect
TSQL

## Version
`sqlfluff, version 0.9.1`

## Configuration
```
[sqlfluff]
dialect = tsql
```

**Comment by barrywhart** [comment]
Using your test SQL and `.sqlfluff` with 0.9.1, I get a parse error:
```
(sqlfluff-3.10.1) ➜  sqlfluff git:(main) ✗ sqlfluff lint test.sql               
== [test.sql] FAIL                                                                                                                                                                                      
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: "exec
                       | function\n @param1 = 'blah',\n @par..."
All Finished 📜 🎉!
```

Can you double check that this is the correct example?

**Comment by fdw** [comment]
I checked the example again, and found something interesting, but I never got an error. The `.sqlfluff` file is exactly as shown, and I made sure to update `sqlfluff` again to the latest version.

Regarding my findings: The pseudo-function-name `function` was a bad idea, as it is parsed differently.
Using these examples
```sql
exec something
  @param1 = 'blah',
  @param2 = null;

exec something
  @param1 = null,
  @param2 = 'blah';

exec function
  @param1 = 'blah',
  @param2 = null;

exec function
  @param1 = null,
  @param2 = 'blah';
```
I would expect all of them to be formatted quite correctly.

However, this is the result `sqlfluff` produces:
```sql
exec something
@param1 is 'blah',
@param2 is null;

exec something
@param1 is null,
@param2 = 'blah';

exec function
  @param1 = 'blah',
  @param2 = null;

exec function
  @param1 = null,
  @param2 = 'blah';
```
As you can see, with a different function name, the indentation is removed and for *some* parameters, `=` is replaced with `is`.

**Comment by fdw** [comment]
Extending my last experiment, it's getting weirder:

Input:
```sql
exec something
  @param1 = 'blah',
  @param2 = 'blah',
  @param3 = null,
  @param4 = 'blah';
```

Output:
```sql
exec something
@param1 is 'blah',
@param2 is 'blah',
@param3 is null,
@param4 = 'blah';
```

Seems to me that `=` is replaced for all parameters up to the `null`, but no further.
